pkg/client/multi.go

```diff
@@ -324,63 +333,69 @@ func (m *MultiClient) VolumeDelete(ctx context.Context, name string) (*apiv1.Vol
 }
 
 func (m *MultiClient) ImageList(ctx context.Context) ([]apiv1.Image, error) {
-	c, err := m.Factory.ForProject(ctx, m.Factory.DefaultProject())
-	if err != nil {
-		return nil, err
-	}
-	return c.ImageList(ctx)
+	return aggregate(ctx, m.Factory, func(c Client) ([]apiv1.Image, error) {
+		return c.ImageList(ctx)
+	})
 }
 
 func (m *MultiClient) ImageGet(ctx context.Context, name string) (*apiv1.Image, error) {
-	c, err := m.Factory.ForProject(ctx, m.Factory.DefaultProject())
+	c, name, err := parseProject(ctx, name, m.Factory)
 	if err != nil {
 		return nil, err
 	}
 	return c.ImageGet(ctx, name)
 }
 
 func (m *MultiClient) ImageDelete(ctx context.Context, name string, opts *ImageDeleteOptions) (*apiv1.Image, error) {
-	c, err := m.Factory.ForProject(ctx, m.Factory.DefaultProject())
+	c, name, err := parseProject(ctx, name, m.Factory)
 	if err != nil {
 		return nil, err
 	}
 	return c.ImageDelete(ctx, name, opts)
 }
 
 func (m *MultiClient) ImagePush(ctx context.Context, tagName string, opts *ImagePushOptions) (result <-chan ImageProgress, err error) {
-	c, err := m.Factory.ForProject(ctx, m.Factory.DefaultProject())
+	c, tagName, err := parseProject(ctx, tagName, m.Factory)
 	if err != nil {
 		return nil, err
 	}
 	return c.ImagePush(ctx, tagName, opts)
 }
 
 func (m *MultiClient) ImagePull(ctx context.Context, name string, opts *ImagePullOptions) (result <-chan ImageProgress, err error) {
-	c, err := m.Factory.ForProject(ctx, m.Factory.DefaultProject())
+	c, name, err := parseProject(ctx, name, m.Factory)
 	if err != nil {
 		return nil, err
 	}
 	return c.ImagePull(ctx, name, opts)
 }
 
 func (m *MultiClient) ImageTag(ctx context.Context, image, tag string) error {
-	c, err := m.Factory.ForProject(ctx, m.Factory.DefaultProject())
+	c, image, err := parseProject(ctx, image, m.Factory)
 	if err != nil {
 		return err
 	}
 	return c.ImageTag(ctx, image, tag)
 }
 
 func (m *MultiClient) ImageDetails(ctx context.Context, imageName string, opts *ImageDetailsOptions) (result *ImageDetails, err error) {
```

Comment:
Question: on this function, should we be looking for and parsing the `project::` prefix?